js/MediaStreamRenderer.ts

```diff
@@ -1,290 +1,293 @@
+		function onResultOK(data: string) {
+			callback(data);
+		}
 
-	// zIndex
-	zIndex = parseFloat(computedStyle.zIndex) || parseFloat(this.element.style.zIndex) || 0;
+		function onResultError() {
+			callback(null);
+		}
 
-	// mirrored (detect "-webkit-transform: scaleX(-1);" or equivalent)
-	if (
-		computedStyle.transform === 'matrix(-1, 0, 0, 1, 0, 0)' ||
-		computedStyle['-webkit-transform'] === 'matrix(-1, 0, 0, 1, 0, 0)'
-	) {
-		mirrored = true;
-	} else {
-		mirrored = false;
+		exec(onResultOK, onResultError, 'iosrtcPlugin', 'MediaStreamRenderer_save', [this.id]);
 	}
 
-	// objectFit ('contain' is set as default value)
-	objectFit = computedStyle.objectFit || 'contain';
+	refresh() {
+		debug('refresh()');
+
+		const elementPositionAndSize = this.getElementPositionAndSize(),
+			computedStyle = window.getComputedStyle(this.element);
+
+		let elementLeft = elementPositionAndSize.left,
+			elementTop = elementPositionAndSize.top,
+			elementWidth = elementPositionAndSize.width,
+			elementHeight = elementPositionAndSize.height;
+
+		// get background color
+		const backgroundColorRgba = computedStyle.backgroundColor
+			.replace(/rgba?\((.*)\)/, '$1')
+			.split(',')
+			.map(function (x) {
+				return x.trim();
+			});
+		backgroundColorRgba[3] = '0';
+		this.element.style.backgroundColor = 'rgba(' + backgroundColorRgba.join(',') + ')';
+		backgroundColorRgba.length = 3;
+
+		// get padding values
+		const paddingTop = parseInt(computedStyle.paddingTop) | 0,
+			paddingBottom = parseInt(computedStyle.paddingBottom) | 0,
+			paddingLeft = parseInt(computedStyle.paddingLeft) | 0,
+			paddingRight = parseInt(computedStyle.paddingRight) | 0;
+
+		// fix position according to padding
+		elementLeft += paddingLeft;
+		elementTop += paddingTop;
+
+		// fix width and height according to padding
+		elementWidth -= paddingLeft + paddingRight;
+		elementHeight -= paddingTop + paddingBottom;
+
+		let videoViewWidth = elementWidth,
+			videoViewHeight = elementHeight;
 
-	// clip
-	if (objectFit === 'none') {
-		clip = false;
-	} else {
-		clip = true;
-	}
+		// visible
+		let visible: boolean;
+		if (computedStyle.visibility === 'hidden') {
+			visible = false;
+		} else {
+			visible = !!this.element.offsetHeight; // Returns 0 if element or any parent is hidden.
+		}
 
-	// borderRadius
-	borderRadius = parseFloat(computedStyle.borderRadius);
-	if (/%$/.test(borderRadius)) {
-		borderRadius = Math.min(elementHeight, elementWidth) * borderRadius;
-	}
+		// opacity
+		const opacity = parseFloat(computedStyle.opacity);
+
+		// zIndex
+		const zIndex =
+			parseFloat(computedStyle.zIndex) || parseFloat(this.element.style.zIndex) || 0;
+
+		// mirrored (detect "-webkit-transform: scaleX(-1);" or equivalent)
+		let mirrored: boolean;
+		if (
+			computedStyle.transform === 'matrix(-1, 0, 0, 1, 0, 0)' ||
+			(computedStyle as any)['-webkit-transform'] === 'matrix(-1, 0, 0, 1, 0, 0)' ||
+			computedStyle.webkitTransform === 'matrix(-1, 0, 0, 1, 0, 0)'
```

Comment:
`-webkit-transform` doesn't appear to be a property on `computedStyle`.  `webkitTransform` is, but it's deprecated.

Reply:
this look for all webkitTransform, "-webkit-transform" and transform to be compaible with old ios webviews....

Reply:
"-webkit-transform" was at some point in ios old versions....